packages/babylon/src/tokenizer/context.js

```diff
@@ -103,8 +103,12 @@ tt.incDec.updateContext = function() {
   // tokExprAllowed stays unchanged
 };
 
-tt._function.updateContext = function() {
-  if (this.curContext() !== types.braceStatement) {
+tt._function.updateContext = function(prevType) {
+  if (
+    this.curContext() !== types.braceStatement ||
```

Comment:
I really don't understand what this is for, I think it was a rough approssimation of the check that you are adjusting in this PR. If it reveals to be unnecessary, can you remove it?